src/phoenix/trace/filter.py

```diff
@@ -0,0 +1,180 @@
+    for source_segment, ast_replacement in {
+        field_name: _ast_replacement(f"span.attributes.get('{field_name}')")
+        for field_name in (
+            getattr(semantic_conventions, variable_name)
+            for variable_name in dir(semantic_conventions)
+            if variable_name.isupper()
+            and variable_name.startswith(("RETRIEVAL", "EMBEDDING", "LLM", "TOOL"))
+        )
+    }.items():
+        yield source_segment, ast_replacement
+        yield "attributes." + source_segment, ast_replacement
+        yield "span.attributes." + source_segment, ast_replacement
+
+
+class _Translator(ast.NodeTransformer):
+    _allowed_fields: Mapping[str, ast.expr] = dict(_allowed_replacements())
+
+    def __init__(self, source: str) -> None:
+        # In Python 3.8, we have to use `ast.get_source_segment(source, node)`.
+        # In Python 3.9, we can use `ast.unparse(node)` instead.
+        self._source = source
+
+    def visit_Attribute(self, node: ast.Attribute) -> Any:
+        source_segment: str = cast(str, ast.get_source_segment(self._source, node))
+        if replacement := self._allowed_fields.get(source_segment):
+            return replacement
+        raise SyntaxError(f"invalid expression: {source_segment}")  # TODO: add details
+
+    def visit_Name(self, node: ast.Name) -> Any:
+        source_segment: str = cast(str, ast.get_source_segment(self._source, node))
+        if replacement := self._allowed_fields.get(source_segment):
+            return replacement
+        raise SyntaxError(f"invalid expression: {source_segment}")  # TODO: add details
+
+    def visit_Constant(self, node: ast.Constant) -> Any:
+        return ast.Name(id="_MISSING", ctx=ast.Load()) if node.value is None else node
+
+
+def _validate_expression(expression: ast.Expression, source: str) -> None:
+    # In Python 3.8, we have to use `ast.get_source_segment(source, node)`.
+    # In Python 3.9, we can use `ast.unparse(node)` instead.
+    if not isinstance(expression, ast.Expression):
+        raise SyntaxError(f"invalid expression: {source}")  # TODO: add details
+    for i, node in enumerate(ast.walk(expression.body)):
+        if i == 0:
+            if isinstance(node, (ast.BoolOp, ast.Compare)):
+                continue
+        elif isinstance(
+            node,
+            (
+                ast.Attribute,
+                ast.BinOp,
+                ast.BoolOp,
+                ast.Compare,
+                ast.Constant,
+                ast.Load,
+                ast.Name,
+                ast.Tuple,
+                ast.List,
+                ast.UnaryOp,
+                ast.boolop,
+                ast.cmpop,
+                ast.operator,
+                ast.unaryop,
+            ),
+        ):
+            continue
+        source_segment: str = cast(str, ast.get_source_segment(source, node))
+        raise SyntaxError(f"invalid expression: {source_segment}")  # TODO: add details
+
+
+class _Missing:
```

Comment:
These _ prefixes are really hindering readability - I don't really feel it's worth it.